src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleDictionary.cs

```diff
@@ -156,5 +152,22 @@ public static partial class JsonSerializer
 #endif
             }
         }
+
+        private static void WriteExtensionData(Utf8JsonWriter writer, ref WriteStackFrame frame)
+        {
+            JsonPropertyInfo jsonPropertyInfo = frame.JsonPropertyInfo;
+
+            DictionaryEntry entry = ((IDictionaryEnumerator)frame.Enumerator).Entry;
+            if (entry.Value is JsonElement)
+            {
+                Debug.Assert(entry.Key is string);
+                Debug.Assert(entry.Value is JsonElement);
+
+                string propertyName = (string)entry.Key;
+                JsonElement jsonElement = (JsonElement)entry.Value;
+
+                jsonElement.WriteAsProperty(propertyName, writer);
+            }
```

Comment:
Else, throw?

Reply:
I think so - assuming we want to be strict here vs. ignore. I'll throw InvalidOperationException and add test. Thanks.

Reply:
Would it be an `ArgumentException` in that case?